device/globals/dialog_instance.gd

```diff
@@ -207,17 +205,14 @@ func anim_finished(anim_name):
 		_queue_free()
 
 func _ready():
-	speech_extension = ProjectSettings.get_setting("escoria/application/speech_suffix")
+	speech_suffix = ProjectSettings.get_setting("escoria/application/speech_suffix")
 	add_to_group("events")
 	if has_node("animation"):
 		get_node("animation").connect("animation_finished", self, "anim_finished")
 	label = get_node("anchor/text")
 
-	finish_with_speech = vm.settings.skip_dialog
-
-	#speech_language = TranslationServer.get_locale().substr(0, 2)
-	speech_language = vm.settings.voice_lang
-	if !(speech_language in speech_locales):
-		speech_language = default_speech_language
+	# Ensure a supported speech locale has been set
+	var speech_locales_path = load(ProjectSettings.get_setting("escoria/application/speech_locales_path")).new()
+	assert(vm.settings.voice_lang in speech_locales_path.speech_locales)
```

Comment:
Maybe this could use `vm.report_errors()` as well?

Reply:
It would generally cause the same effect, of the game crashing, but I figured that this is a case where the game/code would be in such a horribly inconsistent state that it's better to let the programmer be told explicitly.

The use case for disabling crashes on error reporting might be collecting errors in broad sweeps instead of fixing them one by one, or something similar.

I can't think of a strong use case for having an unsupported locale - something that should be a pivotal part of the game design - be a benign thing.

Having argued that case, it's a trivial fix so if you want that to happen, I'll make it happen :D

Reply:
Yeah, I think you're probably right. You should check if speech is enabled though, because if it's not, the developer might not have any speech_locales set, and it shouldn't cause an assertion error in that case.